java/Oleksii-Riabov/src/com/zoolatech/lecture4/tasks/_7/Task7.java

```diff
@@ -0,0 +1,45 @@
+package com.zoolatech.lecture4.tasks._7;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Create a method that accepts a string of digits and after each digit
+ * appends its string representation in parentheses.
+ */
+
+public class Task7 {
+    public static void main(String[] args) {
+        String string = "12345678910";
+        System.out.println(stringRepresentation(string));
+    }
+
+    public static String stringRepresentation(String string) {
+        if (string.matches("[0-9]+") && string.length() > 2) {
```

Comment:
1. why do we have a condition `string.length() > 2`? strings of length 1 or 2 should be also considered as valid strings
2. `string.matches` works longer than `string.length` (since the first method uses regular expressions internally); prefer to define firstly operations that work faster, so they can skip some invalid input

Reply:
"string.length" has been removed.